### PR TITLE
Support IPv6 Urls

### PR DESCRIPTION
**Pull request by Marenz**
I tried running the unittests but somehow they all succeeded even though I deliberately broke one.
Still, the code seems to work with a simple test. I am secretly hoping for a CI that tests for me

### CODE REVIEW

**Comment by s-ludwig** [comment]
The unit test issue is probably due to the split into sub packages. Need to use either `dub test --combined` or `dub test :core` now.

The diff looks good so far. The only thing I wonder is if `m_is_ipv6` should be replaced by a check for `.canFind(":")` in `toString()`, to save space, but more importantly to support setting the `.host` property to an IPv6-address after construction.

**Comment by Marenz** [comment]
Updated to use `canFind` and extended unittest to setting `.host` to a ipv6

File: source/vibe/inet/url.d [hunk 1]
```diff
@@ -92,11 +92,31 @@ struct URL {
 					}
 
 					m_host = str[hs .. si];
-					auto pi = m_host.indexOfCT(':');
-					if(pi > 0) {
-						enforce(pi < m_host.length-1, "Empty port in URL.");
-						m_port = to!ushort(m_host[pi+1..$]);
-						m_host = m_host[0 .. pi];
+
+					long findPort ( string src )
```
Comment (s-ludwig) [inline source/vibe/inet/url.d]: This must be either `sizediff_t` or `auto` for the slice operation below to compile with 32-bits.

**Comment by Marenz** [comment]
Changed return type to `auto`. Rebased on latest master.

**Comment by s-ludwig** [comment]
Okay, thanks!

File: source/vibe/inet/url.d [hunk 2]
```diff
@@ -242,7 +262,14 @@ struct URL {
 			dst.put(password);
 			dst.put('@');
 		}
+
+		import std.algorithm.searching : canFind;
```
Comment (s-ludwig) [inline source/vibe/inet/url.d]: Just noticed that this causes the Travis-CI build to fail. Needs to be `import std.algorithm : canFind` for now.

**Comment by Marenz** [comment]
importing `std.algorithm : canFind` now

**Pull request merged**